src/test/java/uk/gov/hmcts/reform/orgrolemapping/domain/service/DroolPrivateLawStaffOrgRolesTest.java

```diff
@@ -0,0 +1,188 @@
+            containsInAnyOrder(roleNames.toArray()));
+
+        roleAssignments.forEach(r -> {
+            assertTrue(roleCategory.contains(r.getRoleCategory().toString()));
+        });
+
+        roleAssignments.stream().filter(c -> c.getGrantType().equals(GrantType.STANDARD)).toList()
+            .forEach(r -> {
+                assertEquals("PRIVATELAW", r.getAttributes().get("jurisdiction").asText());
+                assertEquals(cap.getPrimaryLocationId(), r.getAttributes().get("primaryLocation").asText());
+            });
+
+        List<Map<String, JsonNode>> list = roleAssignments.stream()
+            .map(RoleAssignment::getAttributes).toList();
+        List<String> workTypesCombined = new ArrayList<>();
+
+        for (Map<String, JsonNode> e : list) {
+            if (e.containsKey("workTypes")) {
+                workTypesCombined.add(e.entrySet().stream()
+                    .filter(a -> a.getKey().equals("workTypes"))
+                    .map(Map.Entry::getValue).findFirst().get().textValue());
+            }
+        }
+        assertThat(workTypesCombined,containsInAnyOrder(workTypes.toArray()));
+    }
+
+    @Test
+    void shouldReturnPrivateLawCtsOrgRolesForRoleId_caseAndTaskSupervisor() {
+        allProfiles.clear();
+        Stream.of(9,3).forEach(roleId -> {
+            CaseWorkerAccessProfile profile = TestDataBuilder.buildUserAccessProfile(roleId + "",
+                "ABA5", false);
+            profile.setCaseAllocatorFlag("Y");
+            profile.setTaskSupervisorFlag("Y");
+            allProfiles.add(profile);
+        });
+        List<RoleAssignment> roleAssignments =
+            buildExecuteKieSession(getFeatureFlags("privatelaw_wa_1_0", true));
+
+        //assertion
+        assertFalse(roleAssignments.isEmpty());
+        assertThat(roleAssignments.stream().map(RoleAssignment::getRoleName).collect(Collectors.toList()),
+            containsInAnyOrder("ctsc-team-leader","task-supervisor", "hmcts-ctsc","case-allocator",
+                "hearing-centre-team-leader", "task-supervisor", "hmcts-admin","case-allocator"));
+        roleAssignments.forEach(roleAssignment -> {
+            if ("ctsc-team-leader".equals(roleAssignment.getRoleName())
+                || "hearing-centre-team-leader".equals(roleAssignment.getRoleName())) {
+                assertEquals("PRIVATELAW",roleAssignment.getAttributes().get("jurisdiction").asText());
+                assertEquals("routine_work,access_requests,hearing_work,applications",
+                    roleAssignment.getAttributes().get("workTypes").asText());
+            }
+            if (RoleCategory.ADMIN.equals(roleAssignment.getRoleCategory())
+                && !roleAssignment.getRoleName().contains("hmcts")) {
+                assertThat(new String[]{"7", "6"},
+                    ArrayMatching.hasItemInArray(roleAssignment.getAttributes().get("region").asText()));
```

Comment:
region id will come only one here. Multiple regions will work only for SSCS. Please check this?